Inc/ST-LIB_HIGH/Protections/Boundary.hpp

```diff
@@ -16,246 +22,224 @@ enum ProtectionType : uint64_t {
+	
 	Boundary(Type lower_boundary, Type upper_boundary): lower_boundary(lower_boundary), upper_boundary(upper_boundary){};
-	Boundary(Type* src, Boundary<Type, Protector> boundary): src(src),lower_boundary(boundary.lower_boundary),upper_boundary(boundary.upper_boundary){}
+	Boundary(Type* src, Boundary<Type, Protector> boundary): 
+	src(src),lower_boundary(boundary.lower_boundary),upper_boundary(boundary.upper_boundary)
+	{
+		format_id = BoundaryInterface::format_look_up.at(type_id<Type>);
+		name.reserve(NAME_MAX_LEN);
+		message = new HeapOrder(uint16_t{111},&format_id,&boundary_type_id,&string_len,&name,&this->lower_boundary,&this->upper_boundary,this->src,
+			&Global_RTC::global_RTC.counter,&Global_RTC::global_RTC.second,&Global_RTC::global_RTC.minute,
+			&Global_RTC::global_RTC.hour,&Global_RTC::global_RTC.day,&Global_RTC::global_RTC.month,&Global_RTC::global_RTC.year);
+	}
 	Boundary(Type* src, Type lower_boundary, Type upper_boundary): src(src), lower_boundary(lower_boundary), upper_boundary(upper_boundary){}
 	bool check_bounds()override{
 		if(*src < lower_boundary || *src > upper_boundary) return false;
 		return true;
 	}
-	int get_string_size()override{
-		return snprintf(nullptr,0,format,to_string(*src).c_str(), to_string(lower_boundary).c_str(), to_string(upper_boundary).c_str());
-	}
-	char* serialize(char* dst)override{
-		sprintf(dst, format, to_string(*src).c_str(), to_string(lower_boundary).c_str(), to_string(upper_boundary).c_str());
-		return dst;
-	}
 };
 
 template<>
 struct Boundary<void, ERROR_HANDLER> : public BoundaryInterface{
 	static constexpr ProtectionType Protector = ERROR_HANDLER;
-	static constexpr const char* format = "\"type\": \"ERROR_HANDLER\", \"data\": \"%s\"";
 	Boundary(void*){}
-	Boundary(void*, Boundary<void,ERROR_HANDLER>){}
+	Boundary(void*, Boundary<void,ERROR_HANDLER>)
+	{
+		error_handler_string.reserve(ERROR_HANDLER_MSG_MAX_LEN);
+		message = new HeapOrder(uint16_t{111},&boundary_type_id,&string_len,&name,&error_handler_string_len,&error_handler_string,
+			&Global_RTC::global_RTC.counter,&Global_RTC::global_RTC.second,&Global_RTC::global_RTC.minute,
+			&Global_RTC::global_RTC.hour,&Global_RTC::global_RTC.day,&Global_RTC::global_RTC.month,&Global_RTC::global_RTC.year);
+	}
 	Boundary() = default;
 	bool check_bounds() override{
 		return not ErrorHandlerModel::error_triggered;
 	}
-	int get_string_size()override{
-		return snprintf(nullptr,0,format,BoundaryInterface::get_error_handler_string().c_str());
-	}
-	char* serialize(char* dst)override{
-		sprintf(dst,format,BoundaryInterface::get_error_handler_string().c_str());
-		return dst;
+	void update_error_handler_message(const char* err_message)override{
+		error_handler_string = err_message;
+		error_handler_string_len = error_handler_string.size();
 	}
+	private:
+		uint8_t boundary_type_id = Protector;
+		string error_handler_string;
+		uint16_t error_handler_string_len;
+		static constexpr uint16_t ERROR_HANDLER_MSG_MAX_LEN = 255;
```

Comment:
why would you set a limit?

Reply:
TCP Ethernet packets can be at most 1458 bytes if you want to avoid segementain, will still work though

Reply:
just to allocate a max size, I can make the buffer as big as I want.